### DISCUSSION TITLE
What is the reason for "NLR jump failed" when calling "mp_call_function_1" in FreeRTOS Task?

### DISCUSSION THREAD

**Discussion by Mars-CN** [micropython/micropython#12614] [ESP32]
I used "esp_event_loop_create_default()" to create an EventLoop, but when calling Micropython's functions through "mp_call_function_1" in the EventLoop trigger function, an error of "NLR jump failed" is reported. It seems that the problem is still in the context. If you call the "mp_call_function_1" function in the Task of FreeRTOS?

**Comment by dlech** [comment]
This probably happens because there is no MicroPython thread running on this RTOS task and therefore nothing to catch unhandled  exceptions.

Depending on what you need to do, one way around this is to schedule a MicroPython function call instead of calling it directly. This was discussed recently in https://github.com/orgs/micropython/discussions/12596.
